src/codec.js

```diff
@@ -52,6 +52,18 @@ function stringToStringTuples (str) {
       throw ParseError('invalid address: ' + str)
     }
 
+    // if it's a path proto, take the rest
+    if (proto.path) {
+      tuples.push([
+        part,
+        // TODO: should we need to check each path part to see if it's a proto?
```

Comment:
Maybe we can go with this now, and when we add new protocols we thing about it?

Reply:
Yeah. It should be reasonable for us to have an address that's `/unix/a/b/c/d/e/f/ip4/0.0.0.0/tcp/1234`, but users would need to avoid addresses like `/unix/c/p2p/d.sock/ip4/0.0.0.0/tcp/12341`, since it has a proto name in the unix path. Having at least basic support for unix addresses is the important thing right now.